File: app/src/main/java/com/example/hades/androidpo/_1_render_op/startup/v1/RxLifeCycleSchedulerBinderImpl.java
package com.example.hades.androidpo._1_render_op.startup.v1;

import android.util.Log;

import org.jetbrains.annotations.NotNull;

import io.reactivex.Observable;
import io.reactivex.ObservableSource;
import io.reactivex.ObservableTransformer;
import io.reactivex.Scheduler;

public class RxLifeCycleSchedulerBinderImpl extends RxLifeCycleBinderImpl implements RxLifeCycleSchedulerBinder {
    private static final String TAG = "RxLifeCycleSchedulerBinderImpl";
    protected Scheduler scheduler;

    public RxLifeCycleSchedulerBinderImpl(Scheduler scheduler) {
        this.scheduler = scheduler;
    }

    // 对Observabl进行变换，
    @Override
    public <T> ObservableTransformer<T, T> bindWithScheduler() {
        return new ObservableTransformer<T, T>() {
            @NotNull
            @Override
            public ObservableSource<T> apply(@NotNull Observable<T> upstream) {
                ObservableSource<T> source = upstream.compose(bind()).observeOn(scheduler);
                Log.e(TAG, "apply:upstream@" + upstream.toString() + ",dest@" + source.hashCode());
                return source;
            }
        };
//        return upstream -> upstream
//                .compose(bind())
//                .observeOn(scheduler);
    }
}

File: app/src/main/java/com/example/hades/androidpo/_1_render_op/startup/v2/RxLifeCycleSchedulerBinderImpl.kt
package com.example.hades.androidpo._1_render_op.startup.v2

import io.reactivex.ObservableTransformer
import io.reactivex.Scheduler

open class RxLifeCycleSchedulerBinderImpl(protected var scheduler: Scheduler) : RxLifeCycleBinderImpl(), RxLifeCycleSchedulerBinder {
    override fun <T> bindWithScheduler(): ObservableTransformer<T, T>? {
        return ObservableTransformer { upstream ->
            upstream.compose(super.bind())
                .observeOn(scheduler)
        }
        //        return upstream -> upstream
//                .compose(bind())
//                .observeOn(scheduler);
    }
}